### ISSUE TITLE
XRDP is logging some normal conditions as ERROR messages

### ISSUE THREAD

**Issue by angelsion** [neutrinolabs/xrdp#1775]
```
[20210102-16:53:50] [ERROR]   out xrdp_mcs_disconnect error - 2
[20210102-16:54:06] [INFO ] Socket 12: AF_INET connection received from 192.168.2.82 port 60503
[20210102-16:54:06] [INFO ] Using default X.509 certificate: /etc/xrdp/cert.pem
[20210102-16:54:06] [INFO ] Using default X.509 key file: /etc/xrdp/key.pem
[20210102-16:54:06] [INFO ] connected client computer name: DESKTOP-IO0GU86
[20210102-16:54:06] [INFO ]   client supports 40 bit encryption
[20210102-16:54:06] [INFO ]   client supports 128 bit encryption
[20210102-16:54:06] [INFO ]   client supports 56 bit encryption
[20210102-16:54:06] [INFO ]   client supports fips encryption
[20210102-16:54:06] [INFO ] adding channel item name rdpdr chan_id 1004 flags 0x80800000
[20210102-16:54:06] [INFO ] adding channel item name rdpsnd chan_id 1005 flags 0xc0000000
[20210102-16:54:06] [INFO ] adding channel item name cliprdr chan_id 1006 flags 0xc0a00000
[20210102-16:54:06] [INFO ] adding channel item name drdynvc chan_id 1007 flags 0xc0800000
[20210102-16:54:06] [ERROR] error unknown xrdp_sec_process_mcs_data tag 0xc006 size 8
[20210102-16:54:06] [ERROR] error unknown xrdp_sec_process_mcs_data tag 0xc00a size 8
[20210102-16:54:06] [INFO ] xrdp_load_keyboard_layout: keyboard_type [4] keyboard_subtype [0]
[20210102-16:54:06] [INFO ] xrdp_load_keyboard_layout: model [] variant [] layout [us] options []
[20210102-16:54:06] [INFO ] TLS connection established from 192.168.2.82 port 60503: TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384
[20210102-16:54:06] [INFO ] xrdp_caps_process_pointer: client supports new(color) cursor
[20210102-16:54:06] [INFO ] xrdp_process_offscreen_bmpcache: support level 1 cache size 10485760 MB cache entries 100
[20210102-16:54:06] [INFO ] xrdp_caps_process_codecs: nscodec, codec id 1, properties len 3
[20210102-16:54:06] [WARN ] xrdp_caps_process_codecs: unknown codec id 5
[20210102-16:54:06] [INFO ] xrdp_caps_process_codecs: RemoteFX, codec id 3, properties len 49
[20210102-16:54:06] [INFO ] Loading keymap file /etc/xrdp/km-00000409.ini
[20210102-16:54:06] [WARN ] local keymap file for 0x00000409 found and doesn't match built in keymap, using local keymap file
[20210102-16:54:06] [ERROR] libxrdp_query_channel - Channel 0 name rdpdr
[20210102-16:54:06] [ERROR] libxrdp_query_channel - Channel 1 name rdpsnd
[20210102-16:54:06] [ERROR] libxrdp_query_channel - Channel 2 name cliprdr
[20210102-16:54:06] [ERROR] libxrdp_query_channel - Channel 3 name drdynvc
[20210102-16:54:06] [ERROR] libxrdp_query_channel - Channel out of range 4
[20210102-16:54:06] [INFO ] drdynvc_process_capability_response: DVC version 3 selected
[20210102-16:54:21] [INFO ] xrdp_wm_log_msg: sesman connect ok
[20210102-16:54:21] [INFO ] xrdp_wm_log_msg: login successful for display 12
[20210102-16:54:21] [ERROR] libxrdp_query_channel - Channel 0 name rdpdr
[20210102-16:54:21] [ERROR] libxrdp_query_channel - Channel 1 name rdpsnd
[20210102-16:54:21] [ERROR] libxrdp_query_channel - Channel 2 name cliprdr
[20210102-16:54:21] [ERROR] libxrdp_query_channel - Channel 3 name drdynvc
[20210102-16:54:21] [ERROR] libxrdp_query_channel - Channel out of range 4
```

**Comment by metalefty** [comment]
If you're not a robot but a human, speak words about the issue you want to solve. Not copy & pasting logs.

**Comment by SH160388** [comment]
I think i have the same issue. And i'am not sure but maybe since kernel upgarde to linux510. (i use manjaro xfce) 
The impact of this issue is that i can't use any mouse or keyboard input. 

[20210105-18:31:34] [WARN ] unknown in xrdp_caps_process_confirm_active 27
[20210105-18:31:34] [WARN ] xrdp_caps_process_confirm_active: not enough orders supported by client, client wants off screen bitmap but offscreen bitmaps disabled
[20210105-18:31:34] [INFO ] Loading keymap file /etc/xrdp/km-00000407.ini
[20210105-18:31:34] [WARN ] local keymap file for 0x00000407 found and doesn't match built in keymap, using local keymap file
[20210105-18:31:34] [ERROR] libxrdp_query_channel - Channel 0 name RDPDR
[20210105-18:31:34] [ERROR] libxrdp_query_channel - Channel 1 name rdpsnd
[20210105-18:31:34] [ERROR] libxrdp_query_channel - Channel 2 name cliprdr
[20210105-18:31:34] [ERROR] libxrdp_query_channel - Channel 3 name drdynvc
[20210105-18:31:34] [ERROR] libxrdp_query_channel - Channel out of range 4
[20210105-18:31:35] [INFO ] xrdp_wm_log_msg: sesman connect ok
[20210105-18:31:35] [INFO ] drdynvc_process_capability_response: DVC version 2 selected
[20210105-18:31:35] [INFO ] xrdp_wm_log_msg: login successful for display 10

**Comment by matt335672** [comment]
In the absence of any other information from the OP, I'm going to assume that this issue means the following:-

1. This line in `libxrdp/xrdp.c` should be logging a DEBUG message instead on an ERROR message:-

https://github.com/neutrinolabs/xrdp/blob/c76c58044119e6fc2bbe7fa6e9819448e07582ea/libxrdp/libxrdp.c#L1196

2. This logic in `xrdp/xrdp_mm.c `

https://github.com/neutrinolabs/xrdp/blob/c76c58044119e6fc2bbe7fa6e9819448e07582ea/xrdp/xrdp_mm.c#L662-L670

needs to be modified somehow to not call `libxrdp_get_channel_id()` with an invalid index.

The first of these is being addressed by PR #1742 

I'm changing the title of this fault report to match my assumptions above.

@angelsion - if this isn't your intention, please feel free to open another issue which describes your problem clearly.

@SH160388 - you've clearly got here with an internet search for an error message. Feel free to open a separate issue which describes your problem clearly.
